halo2_gadgets/src/utilities/lookup_range_check.rs

```diff
@@ -542,34 +649,131 @@ mod tests {
             }
         }
 
+        // Read the old proofs into 'proofs'
+        let proofs = {
+            let test_case_bytes =
+                fs::read("src/utilities/circuit_proof_test_case_short_range_check.bin").unwrap();
+            read_all_proofs(&test_case_bytes[..], 1888).expect("proof must be valid")
+        };
+
+        // Setup phase: generate parameters
+        let params: Params<Affine> = Params::new(11);
+
         // Edge case: zero bits
+        // case 0
         {
             let circuit: MyCircuit<pallas::Base> = MyCircuit {
                 element: Value::known(pallas::Base::ZERO),
                 num_bits: 0,
             };
             let prover = MockProver::<pallas::Base>::run(11, &circuit, vec![]).unwrap();
             assert_eq!(prover.verify(), Ok(()));
+
+            // generate vk
+            let vk = plonk::keygen_vk(&params, &circuit).unwrap();
+
+            // Test that the pinned verification key (representing the circuit)
+            // is as expected. Which indicates the layouters are the same.
+            {
+                //panic!("{:#?}", vk.pinned());
+                assert_eq!(
+                    format!("{:#?}\n", vk.pinned()),
+                    include_str!("vk_short_range_check_0").replace("\r\n", "\n")
+                );
+            }
+
+            // serialized_proof_test_case
+            {
+                match proofs.get(0) {
+                    Some(proof) => {
+                        // Verify the old proofs[0] with the new vk
```

Comment:
`// Verify the stored proof (case 0) against the generated vk` (all occurrences)